test/integration/apiserver/portforward/portforward_test.go

```diff
@@ -113,7 +113,15 @@ func TestPortforward(t *testing.T) {
 	// local port missing asks os to find random open port.
 	// Example: ":8000" (local = random, remote = 8000)
 	localRemotePort := fmt.Sprintf(":%s", remotePort)
-	streams, _, out, errOut := genericiooptions.NewTestIOStreams()
+	out := &mBuffer{
```

Comment:
There are quite a few other tests which use `NewTestIOStreams`. That method is marked "deprecated" with itself (?!) as replacement. I haven't looked into the history.

Would it perhaps make sense to fix that method so that it returns thread-safe streams?

Otherwise we might have to adapt the other tests at some future time....

Reply:
I also got confused for a moment, looks like there are two similarly named packages that expose `NewTestIOStreams`, `genericiooptions` and `genericiclioptions`.

Reply:
> Would it perhaps make sense to fix that method so that it returns thread-safe streams?

IIRC the streams are not safe because they are used on tests and the buffers are shared, same as the test on the apiserver that need to do deepcopy on the vars